compiler/src/dotty/tools/dotc/reporting/diagnostic/Message.scala

```diff
@@ -56,6 +86,11 @@ abstract class Message(val errorId: ErrorMessageID) { self =>
     */
   def explanation: String
 
+  /** Links may list URLs to Internet resources related to the error
+    * e.g. the Scala Language Specification.
+    */
+  val links: List[DocumentationLink] = Nil
```

Comment:
That would mean every error message had to implement it, right now few errors contain links. It might be an idea to have it, though -  so that all errors try point to something in the Language Spec, but that's a lot of work.

Reply:
I get the intention, but overrides are code smell to me...

If the diff is not too huge it might also be fine to

- Simply add `val links = Nil` to those that don't have anything

- Refactor Message into a case class with default argument (IDK why it wasn't made a case class in the first place?)

- Use inheritance, where `LinkedMessage` is to be used instead of `Message` when they are links:
   ```scala
  class LinkedMessage { def links }
  class Message extends LinkedMessage { def links = Nil }
  ```

I'm just pointing out a small detail, if you think overrides are better than the above then it's find to keep it that was.